### PR TITLE
OCM-2666: Remove AWS validations from rosa_cluster resource

### PR DESCRIPTION
**Pull request by nirarg**
This change is done in order to avoid the requirement to have valid AWS credentials in the local machine, in which the terraform executed. This change include:
* Remove the relevant code from the provider
* changing the subsystem accordingly to the change
* Removing limitation from docs

### CODE REVIEW

File: provider/cluster_rosa_classic_resource.go [hunk 1]
```diff
@@ -1126,6 +900,7 @@ func (r *ClusterRosaClassicResource) Create(ctx context.Context,
 		)
 		return
 	}
+	tflog.Info(ctx, fmt.Sprintf("************************ 1"))
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:903]: please remove the log prints

File: provider/cluster_rosa_classic_resource.go [hunk 2]
```diff
@@ -1138,6 +913,7 @@ func (r *ClusterRosaClassicResource) Create(ctx context.Context,
 		)
 		return
 	}
+	tflog.Info(ctx, fmt.Sprintf("************************ 2"))
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:916]: please remove the log prints

File: provider/common/helpers.go [hunk 3]
```diff
@@ -151,3 +151,12 @@ func HandleErr(res *ocmerrors.Error, err error) error {
 	errType := weberr.ErrorType(res.Status())
 	return errType.Set(errors.Errorf("%s", msg))
 }
+
+// StringValue returns the value of the string pointer passed in or
+// "" if the pointer is nil.
```
Comment (bardielle) [inline provider/common/helpers.go:156]: If I'm not wrong - that function is not needed anymore...

Comment (nirarg) [reply provider/common/helpers.go]: You right, removed it

File: provider/cluster_rosa_classic_resource.go (hunk 1 above)
Comment (nirarg) [reply provider/cluster_rosa_classic_resource.go]: Oops ... removed

File: provider/cluster_rosa_classic_resource.go (hunk 2 above)
Comment (nirarg) [reply provider/cluster_rosa_classic_resource.go]: Oops ... removed

**Pull request merged**